plugins/inputs/modbus/configuration_request.go

```diff
@@ -42,6 +46,17 @@ func (c *ConfigurationPerRequest) Check() error {
 	seenFields := make(map[uint64]bool)
 
 	for _, def := range c.Requests {
+		// Check for valid optimization
+		validOptimizations := []string{"", "none", "shrink", "rearrange", "aggressive", "max_insert"}
+		if !choice.Contains(def.Optimization, validOptimizations) {
+			return fmt.Errorf("unknown optimization %q", def.Optimization)
+		}
+		if def.Optimization == "max_insert" && def.MaxExtraRegisters > maxQuantityHoldingRegisters {
+			return fmt.Errorf("max_insert optimization requires max_extra_registers to be smaller than %v but %v was provided", maxQuantityHoldingRegisters, def.MaxExtraRegisters)
+		}
+		if def.Optimization == "max_insert" && def.MaxExtraRegisters == 0 {
+			return fmt.Errorf("max_insert optimization requires max_extra_registers to be larger than 0 but %v was provided or was not set", def.MaxExtraRegisters)
+		}
```

Comment:
Maybe this would be nicer to read and more extensible in the future
```suggestion
		// Check for valid optimization
		switch def.Optimization {
		case "", "none", "shrink", "rearrange", "aggressive":
		case "max_insert":
			if def.MaxExtraRegisters <= 0 || def.MaxExtraRegisters > maxQuantityHoldingRegisters {
				return fmt.Errorf("max_extra_registers has to be between 1 and %d", maxQuantityHoldingRegisters,)
			}
		default:
			return fmt.Errorf("unknown optimization %q", def.Optimization)
		}
```